lib/resty/jwt.lua

```diff
@@ -91,10 +91,15 @@ _M.alg_whitelist = nil
 function _M.sign(self, secret_key, jwt_obj)
     -- header typ check
     local typ = jwt_obj["header"]["typ"]
-    if typ ~= "JWT" then
-        error({reason="invalid typ: " .. typ})
+    -- Optional header typ check [See http://tools.ietf.org/html/draft-ietf-oauth-json-web-token-25#section-5.1]
+    if jwt_obj["header"]["typ"] ~= nil then 
```

Comment:
line 93 declared variable typ
maybe it's better to use `tye ~= nil and typ ~= "JWT" ` for checking?